src/main.rs

```diff
@@ -160,24 +179,63 @@ async fn write_state(infile: &str, outfile: &str, cachefile: &str) {
     geo_cache.save().await.expect("could not save geoip cache");
 
     let joutput = serde_json::to_string(&state).unwrap();
-    fs::write(outfile, joutput).unwrap();
+    fs::write(config.state_file_path.as_ref().unwrap(), joutput).unwrap();
 }
 
 #[tokio::main]
 async fn main() {
-    let args: Vec<String> = env::args().collect();
+    let arg_conf = ArgConfiguration::parse();
+    let mut configuration: CrunchyConfiguration;
 
-    // TODO(asmie): could be refactored to use eg. clap crate.
-    if args.len() != 4 {
-        println!("\n\nUsage is: cargo run <in-sample.json> <out-state.json> <geoip-cache.json> ");
-        println!("E.g.:     \x1b[93mcargo run --release testdata/sample.json testdata/state.json\x1b[0m\n");
-        return;
+    configuration = arg_conf
```

Comment:
the whole configuration setup could be moved to a separate function

Reply:
Totally agree. 
And as stated in the TODO: https://github.com/runziggurat/crunchy/blob/d29427f6f202ae394e1c32cabfb187691e067585/src/main.rs#L72 it should be done at some point of time.

Reply:
> During the refactorization (previous comment) that should be taken out from here

but I must be honest, this piece of software was not easy to read - it took me more than I expected to understand what was going on to properly review it

Reply:
even the previous PR, which I tried to review, but I couldn't foresee how it would fit the future code

maybe we could have a short description of the code structure that'll be delivered later - so the reviewer could understand all the pieces will fit together